app/src/main/java/com/quran/labs/androidquran/ui/adapter/TranslationsAdapter.java

```diff
@@ -152,19 +219,75 @@ ImageView getRightImage() {
       return rightImage;
     }
 
-    final View.OnClickListener removeListener = v -> {
-      TranslationItem item = (TranslationItem) translations.get(getAdapterPosition());
-      onClickRemoveSubject.onNext(item);
+    final View.OnLongClickListener actionMenuListener = v -> {
+      if (actionMode != null) {
+        actionMode.finish();
+        selectionListener.clearSelection();
+      } else if (activity != null) {
+        selectionListener.handleSelection(item);
+        actionMode = activity.startSupportActionMode(new ModeCallback());
+      }
+      return true;
     };
 
     @Override
     public void onClick(View v) {
-      TranslationItem item = (TranslationItem) translations.get(getAdapterPosition());
-      if (item.exists() && !item.needsUpgrade()) {
-        onClickRemoveSubject.onNext(item);
-      } else {
+      if (actionMode != null) {
+        actionMode.finish();
+      }
+      selectionListener.clearSelection();
+      if (!item.exists() || item.needsUpgrade()) {
         onClickDownloadSubject.onNext(item);
       }
     }
   }
+
+  private class ModeCallback implements ActionMode.Callback  {
+    @Override
+    public boolean onCreateActionMode ( ActionMode mode, Menu menu )
```

Comment:
Please consider following the existing coding style.

Reply:
@ozbek I have not figured out how project's code style works yet... is there an actual formatter config I am overlooking?  Or is there just something(s) specific that you'd like to stick to?

Reply:
Nothing serious, it's just how you have put the braces and use of white-space.